### ISSUE TITLE
High Vulnerability Security Issue

### ISSUE THREAD

**Issue by umuttechin** [brianc/node-postgres#3138]
Hi,

We have been using pg-promise v11.5.4 for a Typescript project. Since 2024.01.13 we have been having following issues:

```
2024-01-29 00:10:13 UTC INFO  [main] --- Policy rule "High Vulnerability Security Issue" was violated by component "node_postgres:1.1.1" (https://*************************/api/components/004e0535-3fc4-4b69-a943-dfbc61df9517/versions/763e60b6-4bde-4f1e-bcd8-73e84fd18a09).
2024-01-29 00:10:13 UTC INFO  [main] --- Policy rule "High Vulnerability Security Issue" has a fatal severity type: Blocker with description: High Vulnerability security issue, with a CVSS score greater than or equal to 7.0
```

Because, pg-promise depends node-postgres it fails. also. However, we don' t have direct access to container to check the issue elaborately.

We have tried different pg-promise versions, but they also failed with the same error(`Policy rule "High Vulnerability Security Issue" was violated by component "node_postgres:1.1.1"`), so that we assumed pg-promise uses a fixed version of node-postgres. We were right because according to the [case](https://github.com/vitaly-t/pg-promise/issues/908) we opened for pg-promise, pg-promise uses the latest version of underlying driver which node-postgres.

Are you aware of this situation or is there any plan in your calendar to fix it?

If you need any details please let me know.

Dockerfile:
```

FROM amazon/aws-lambda-nodejs:16
WORKDIR ${LAMBDA_TASK_ROOT}

COPY package.json ./
RUN npm install
COPY *.js ./

CMD [ "index.handler" ]
```

Thanks!

**Comment by charmander** [comment]
> Are you aware of this situation or is there any plan in your calendar to fix it?

I’m not aware of any published vulnerabilities in the current version of pg (@brianc?). The handful of Google results (3) I can find for `"Policy rule" "was violated by component"` seem related to some “Synopsys” product. 1.1.1 isn’t going to be the version of this package (unless you’ve accidentally introduced an old version of pg as a dependency – check `npm ls pg` in any case), so it’s not clear which vulnerability it’s referring to (have you looked at that censored link?).

> If you need any details please let me know.

Yes, we need any details.

**Comment by umuttechin** [comment]
Hi,

This is the whole relevant part of the log for this vulnerability:

```
2024-01-18 00:11:31 UTC INFO  [main] --- Overall Policy Status of project version as reported by Black Duck: IN_VIOLATION
2024-01-18 00:11:31 UTC INFO  [main] --- Black Duck found: 2 components in violation (Policy Severity counts: 0 matches have a severity level of UNSPECIFIED, 0 matches have a severity level of TRIVIAL, 0 matches have a severity level of OK, 0 matches have a severity level of MINOR, 1 match has a severity level of MAJOR, 0 matches have a severity level of CRITICAL, 1 match has a severity level of BLOCKER), 0 components in violation, but overridden, and 97 components not in violation.
2024-01-18 00:11:31 UTC INFO  [main] --- Searching BOM for components in violation of fatal policy rules.
2024-01-18 00:11:37 UTC INFO  [main] --- Policy rule "High Vulnerability Security Issue" was violated by component "node_postgres:1.1.1" (https://*************************/api/components/004e0535-3fc4-4b69-a943-dfbc61df9517/versions/763e60b6-4bde-4f1e-bcd8-73e84fd18a09).
2024-01-18 00:11:37 UTC INFO  [main] --- Policy rule "High Vulnerability Security Issue" has a fatal severity type: Blocker with description: High Vulnerability security issue, with a CVSS score greater than or equal to 7.0
```

Here is the out put of `npm ls pg`

```
bash-4.2# npm ls pg
rds-trigger-handler@1.0.0 /var/task
└─┬ pg-promise@11.5.4
  └─┬ pg@8.11.3
    └─┬ pg-pool@3.6.1
      └── pg@8.11.3 deduped
```

So, we use latest version of pg.

Also, we searched for this specific vulnerability on Google, but did not find anything useful up to now.

Thanks!

**Comment by charmander** [comment]
Did you open the link it gives you?

> "node_postgres:1.1.1" (https://\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*/api/components/004e0535-3fc4-4b69-a943-dfbc61df9517/versions/763e60b6-4bde-4f1e-bcd8-73e84fd18a09)

**Comment by umuttechin** [comment]
Hi,

We can' t open it, because it is also masked for us too.

I will try to reach out blackduck console manager to if we are able to and update the case.

**Comment by umuttechin** [comment]
Hi,

I got access to Blackduck portal and found the vulnerability ID which is CVE-2017-16082.

It seems there is no short or long term remedy for this particular case.

**Comment by sehrope** [comment]
That CVE is from 2017 and has was fixed by this driver's maintainer immediately after it was reported (by me!). The CVE detail page you linked to lists out the affected versions and they do not intersect with the versions in your `npm ls pg` output.

**Comment by charmander** [comment]
Yeah. This is probably a false positive, but not a pg issue in any case.

**Comment by vishal-choudhary-bh** [comment]
+1 our black duck scan also resulted in CVE-2017-16082 with 8.11.3 version. we are also assuming it to be false positive. can the repo owner own this and clear this from NVD database : https://nvd.nist.gov/vuln/detail/CVE-2017-16082.

**Comment by charmander** [comment]
@vishal-choudhary-bh What information in that database is wrong?

**Comment by vishal-choudhary-bh** [comment]
its still showing as an open issue in the DB and not as mitigated which is letting black duck believe that the issues is still active.

**Comment by janpio** [comment]
Doesn't this list exactly which versions were vulnerable, which implies that all others (including 8.11.3) are not?

<img width="606" alt="image" src="https://github.com/brianc/node-postgres/assets/183673/668f1a8c-0613-4dba-b9d2-b078c38f3a5f">
